dingo/backup/redundant_facet_removal_backup.py

```diff
@@ -0,0 +1,392 @@
+
+    # Optimize the updated model
+    model.optimize()
+
+    return model
+
+def fast_remove_redundant_facets(lb, ub, S, c, opt_percentage=100):
+    if lb.size != S.shape[1] or ub.size != S.shape[1]:
+        raise Exception(
+            "The number of reactions must be equal to the number of given flux bounds."
+        )
+
+    redundant_facet_tol = 1e-07
+    tol = 1e-06
+
+    m = S.shape[0]
+    n = S.shape[1]
+    beq = np.zeros(m)
+    Aeq_res = S
+
+    A = np.zeros((2 * n, n), dtype="float")
+    A[0:n] = np.eye(n)
+    A[n:] -= np.eye(n, n, dtype="float")
+
+    b = np.concatenate((ub, -lb), axis=0)
+    b = np.asarray(b, dtype="float")
+    b = np.ascontiguousarray(b, dtype="float")
+
+    max_biomass_flux_vector, max_biomass_objective = fast_fba(lb, ub, S, c)
+    val = -np.floor(max_biomass_objective / tol) * tol * opt_percentage / 100
+
+    b_res = []
+    A_res = np.empty((0, n), float)
+    beq_res = np.array(beq)
+
+    try:
+        with gp.Env(empty=True) as env:
+            env.setParam("OutputFlag", 0)
+            env.start()
+
+            with gp.Model(env=env) as model:
+                x = model.addMVar(
+                    shape=n,
+                    vtype=GRB.CONTINUOUS,
+                    name="x",
+                    lb=lb,
+                    ub=ub,
+                )
+
+                Aeq_sparse = sp.csr_matrix(S)
+                A_sparse = sp.csr_matrix(np.array(-c))
+                b_sparse = np.array(val)
+
+                b = np.array(b)
+                beq = np.array(beq)
+
+                model.addMConstr(Aeq_sparse, x, "=", beq, name="c")
+                model.update()
+                model.addMConstr(A_sparse, x, "<", [val], name="d")
+                model.update()
+
+                model_iter = model.copy()
+
+                indices_iter = range(n)
+                removed = 1
+                offset = 1
+                facet_left_removed = np.zeros((1, n), dtype=bool)
+                facet_right_removed = np.zeros((1, n), dtype=bool)
+
+                while removed > 0 or offset > 0:
+                    removed = 0
+                    offset = 0
+                    indices = indices_iter
+                    indices_iter = []
+
+                    Aeq_sparse = sp.csr_matrix(Aeq_res)
+                    beq = np.array(beq_res)
+
+                    b_res = []
+                    A_res = np.empty((0, n), float)
+                    for i in indices:
+                        objective_function = A[i]
+
+                        redundant_facet_right = True
+                        redundant_facet_left = True
+
+                        objective_function_max = np.asarray(
+                            [-x for x in objective_function]
+                        )
+
+                        # Call solve_lp_with_different_objectives to solve LP
+                        model_iter = solve_lp_with_different_objectives(
+                            model_iter, objective_function_max
```

Comment:
This has to be:
model_iter = solve_lp_with_different_objectives(
                            model_iter.copy(), objective_function_max
                        )

Plz change this wherever it has

Reply:
ok